flood_adapt/dbs_controller.py

```diff
@@ -968,12 +969,27 @@ def get_measures(self) -> dict[str, Any]:
         objects = [MeasureFactory.get_measure_object(path) for path in measures["path"]]
         measures["name"] = [obj.attrs.name for obj in objects]
         measures["long_name"] = [obj.attrs.long_name for obj in objects]
-        measures["geometry"] = [
-            gpd.read_file(path.parent.joinpath(obj.attrs.polygon_file))
-            if obj.attrs.polygon_file is not None
-            else None
-            for (path, obj) in zip(measures["path"], objects)
-        ]
+
+        geometries = []
+        for path, obj in zip(measures["path"], objects):
+            # If polygon is used read the polygon file
+            if obj.attrs.polygon_file is not None:
+                geometries.append(
+                    gpd.read_file(path.parent.joinpath(obj.attrs.polygon_file))
+                )
+            # If aggregation area is used read the polygon from the aggregation area name
+            elif obj.attrs.aggregation_area_name is not None:
+                # TODO Maybe it makes more sense to have the geometry returned here?
+                aggr_areas = self.get_aggregation_areas()
```

Comment:
Running this once before the loop would already help a lot I assume in speeding up the code. You don't need to call it every loop. get_aggregation_areas() is a very slow function since it reads the entire file and converts all aggregation areas to another crs every time you run